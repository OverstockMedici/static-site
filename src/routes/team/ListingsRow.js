import React from 'react';
import { StyledArrow } from './team.style';

const arrowRight = '/img/arrow-right-green.png';

const ListingsRow = ({ title, url }) => {
    return (
        <div data-testid="listings-row" key={title} className="job-listing">
            <a href={url} target="_blank" rel="noopener noreferrer">
                <h3>{title}</h3>
            </a>
            <div className="job-description">
                <a
                    href={url}
                    title="Read Full Job Description"
                    target="_blank"
                    rel="noopener noreferrer"
                >
                    <StyledArrow src={arrowRight} alt="View Job" />
                </a>
            </div>
        </div>
    );
};

export default ListingsRow;
