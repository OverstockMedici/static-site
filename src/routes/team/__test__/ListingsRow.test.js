import React from 'react';
import ReactDOM from 'react-dom';
import ListingsRow from './../ListingsRow';

import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import renderer from 'react-test-renderer';

afterEach(cleanup);

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ListingsRow />, div);
});

it('renders ListingsRow correctly', () => {
    const { getByTestId, container } = render(
        <ListingsRow
            title="Pro Esports Player"
            url="https://www.fnatic.com/"
        />,
    );
    expect(getByTestId('listings-row')).toHaveTextContent('Pro Esports Player');
    expect(document.querySelector('a').getAttribute('href')).toBe(
        'https://www.fnatic.com/',
    );
});

it('matches snapshot', () => {
    const listingsRowTree = renderer
        .create(
            <ListingsRow
                title="Pro Esports Player"
                url="https://www.fnatic.com/"
            />,
        )
        .toJSON();

    expect(listingsRowTree).toMatchSnapshot();
});
