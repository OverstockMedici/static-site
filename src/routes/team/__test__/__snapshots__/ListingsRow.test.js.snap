// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="job-listing"
  data-testid="listings-row"
>
  <a
    href="https://www.fnatic.com/"
    rel="noopener noreferrer"
    target="_blank"
  >
    <h3>
      Pro Esports Player
    </h3>
  </a>
  <div
    className="job-description"
  >
    <a
      href="https://www.fnatic.com/"
      rel="noopener noreferrer"
      target="_blank"
      title="Read Full Job Description"
    >
      <img
        alt="View Job"
        className="teamstyle__StyledArrow-sc-1tz4au3-6 dEKuGS"
        src="/img/arrow-right-green.png"
      />
    </a>
  </div>
</div>
`;
